modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -205,14 +241,16 @@ public StickyConnectionSupplier(Object connection) {
       this.connection = connection;
     }
 
-    @Override
     public Object getConnection() throws MuleException {
       return connection;
     }
 
-    @Override
     public void close() {
 
     }
+
+    public void invalidateConnection() {
+
```

Comment:
there's no point in this method being empty. You need to reconnect here as well and add tests for the sticky connection case